src/transform/src/canonicalization/flat_map_elimination.rs

```diff
@@ -7,16 +7,19 @@
 // the Business Source License, use of this software will be governed
 // by the Apache License, Version 2.0.
 
-//! Turns `FlatMap` into `Map` if only one row is produced by flatmap.
-//!
+//! For a `FlatMap` whose args are all constants, turns it into `Map` if only 1 row is produced by
```

Comment:
Nit but I think args should either be `FlatMap::args` or be spelled out. There is an `args` member of the `FlatMap` stage, and that is what we  are talking about, but you need to know that and not worry about the fact that the main argument of the flatmap, the relation input, doesn't have to be a constant.